foreign_cc/built_tools/cmake_build.bzl

```diff
@@ -1,46 +1,33 @@
 """ Rule for building CMake from sources. """
 
-load("//foreign_cc/private:detect_root.bzl", "detect_root")
-load("//foreign_cc/private:shell_script_helper.bzl", "convert_shell_script")
-
-def _cmake_tool(ctx):
-    root = detect_root(ctx.attr.cmake_srcs)
+load(
+    "//foreign_cc/built_tools/private:built_tools_framework.bzl",
+    "FOREIGN_CC_BUILT_TOOLS_ATTRS",
+    "FOREIGN_CC_BUILT_TOOLS_HOST_FRAGMENTS",
+    "built_tool_rule_impl",
+)
 
-    cmake = ctx.actions.declare_directory("cmake")
+def _cmake_tool_impl(ctx):
     script = [
-        "export BUILD_DIR=##pwd##",
-        "export BUILD_TMPDIR=$${BUILD_DIR}$$.build_tmpdir",
-        "##copy_dir_contents_to_dir## ./{} $BUILD_TMPDIR".format(root),
-        "##mkdirs## " + cmake.path,
-        "cd $$BUILD_TMPDIR$$",
-        "./bootstrap --prefix=install",
+        "./bootstrap --prefix=$$INSTALLDIR$$",
+        # TODO: Use make from a toolchain
+        "make",
         "make install",
-        "##copy_dir_contents_to_dir## ./install $BUILD_DIR/" + cmake.path,
-        "cd $$BUILD_DIR$$",
     ]
```

Comment:
This could probably be rewritten quite easily as a `configure_make` target?

Reply:
Yeah, it can and should since it clearly requires a `make` binary. It'd be better to use a rule which set all that up. Though, I don't know if all the provider info is relevant from the configure rule, I'll have to think about that more.